elephant/spike_train_correlation.py

```diff
@@ -246,3 +249,408 @@ def __calculate_correlation_or_covariance(binned_sts, binary, corrcoef_norm):
+                        "The window exceed the length of the spike trains")
+                # Assign left and right edges of the cch
+                l, r = win[0], win[1]
+            # Window parameter given in time units
+            else:
+                # Check the window parameter values
+                if win[0].rescale(binsize.units).magnitude % \
+                    binsize.magnitude != 0 or win[1].rescale(
+                        binsize.units).magnitude % binsize.magnitude != 0:
+                    raise ValueError(
+                        "The window has to be a multiple of the binsize")
+                if win[0] >= win[1] or win[0] <= -max_num_bins * binsize \
+                        or win[1] >= max_num_bins * binsize:
+                    raise ValueError("The window exceed the length of the"
+                                     " spike trains")
+                # Assign left and right edges of the cch
+                l, r = int(win[0].rescale(binsize.units) / binsize), int(
+                    win[1].rescale(binsize.units) / binsize)
+
+            # Cross correlate the spike trains
+            corr = np.correlate(st2_arr, st1_arr, mode='full')
+            counts = corr[len(st1_arr)+l+1:len(st1_arr)+1+r+1]
+
+        # Case generic
+        else:
+            # Cross correlate the spike trains
+            counts = np.correlate(st2_arr, st1_arr, mode=mode)
+            # Assign the edges of the cch for the different mode parameters
+            if mode == 'full':
+                # Assign left and right edges of the cch
+                r = st_2.num_bins - 1
+                l = - st_1.num_bins + 1
+            # cch compute only for the entries that completely overlap
+            elif mode == 'valid':
+                # Assign left and right edges of the cch
+                r = max(st_2.num_bins - st_1.num_bins, 0)
+                l = min(st_2.num_bins - st_1.num_bins, 0)
+        bin_ids = np.r_[l:r + 1]
+
+        # Correct the values taking into account lacking contributes
+        # at the edges
+        if border_corr is True:
+            correction = float(max_num_bins + 1) / np.array(
+                max_num_bins + 1 - abs(
+                    np.arange(l, r + 1)), float)
+            counts = counts * correction
+
+        # Define the kern for smoothing as an ndarray
+        if hasattr(kern, '__iter__'):
+            if len(kern) > np.abs(l) + np.abs(r) + 1:
+                raise ValueError(
+                    'The length of the kernel cannot be larger than the '
+                    'length %d of the resulting CCH.' % (
+                        np.abs(l) + np.abs(r) + 1))
+            kern = np.array(kern, dtype=float)
+            kern = 1. * kern / sum(kern)
+        # Check kern parameter
+        elif kern is not None:
+            raise ValueError('Invalid smoothing kernel.')
+
+        # Smooth the cross-correlation histogram with the kern
+        if kern is not None:
```

Comment:
some of the code is duplicated in both methods. Could you refactor it into separate functions?